flight/PiOS/Common/pios_ms5xxx.c

```diff
@@ -0,0 +1,698 @@
+				sens -= (7 * (tmp_2000 * tmp_2000)) >> 3;
+
+				// It is correct to nest this if(), as per the docs
+				if (temperature < -1500) {
+					int32_t tmp_1500 = temperature + 1500;
+					sens -= 2 * (tmp_1500 * tmp_1500);
+				}
+			} else if (temperature > 4500) {
+				int32_t tmp_4500 = (temperature - 4500);
+				sens += (tmp_4500*tmp_4500) >> 3;
+			}
+			break;
+		case PIOS_MS5M_MS5803_02:
+			offset = ((int64_t)dev->calibration[1] << 17) + (((int64_t)dev->calibration[3] * delta_temp) >> 6);
+			sens   = ((int64_t)dev->calibration[0] << 16) + (((int64_t)dev->calibration[2] * delta_temp) >> 7);
+
+			// second order temperature compensation
+			if (temperature < 2000) {
+				int32_t tmp_2000 = temperature - 2000;
+				offset -= (61 * (tmp_2000 * tmp_2000)) >> 4;
+				sens   -= ( 2 * (tmp_2000 * tmp_2000)) >> 0;
+
+				// It is correct to nest this if(), as per the docs
+				if (temperature < -1500) {
+					int32_t tmp_1500 = temperature + 1500;
+					offset -= 20 * (tmp_1500 * tmp_1500);
+					sens   -= 12 * (tmp_1500 * tmp_1500);
+				}
+			}
+			break;
+		}
+
+		dev->pressure = ((((int64_t)raw_pressure * sens) >> 21) - offset) >> 15;
+	}
+}
+
+/**
+* Reads one or more bytes into a buffer
+* \param[in] the command indicating the address to read
+* \param[out] buffer destination buffer
+* \param[in] len number of bytes which should be read
+* \return 0 if operation was successful
+* \return -1 if dev is invalid
+* \return -2 if error during I2C transfer
+*/
+static int32_t PIOS_MS5XXX_Read(uint8_t address, uint8_t *buffer, uint8_t len)
+{
+	if (PIOS_MS5XXX_Validate(dev) != 0) {
+		return -1;
+	}
+
+	int32_t rc = 0;
+
+	switch (dev->pios_bus_type) {
+	case PIOS_SPI_BUS:
+	{
+		if (PIOS_MS5XXX_ClaimSPIBus() != 0) {
+			return -2;
+		}
+
+		if (PIOS_SPI_TransferByte(dev->dev_id, address) < 0) {
+			rc = -3;
+			goto out;
+		}
+
+		if (PIOS_SPI_TransferBlock(dev->dev_id, NULL, buffer, len, NULL) < 0) {
+			rc = -3;
+			goto out;
+		}
+
+	out:
+		PIOS_MS5XXX_ReleaseSPIBus();
+		break;
+	}
+	case PIOS_I2C_BUS:
+	{
+		const struct pios_i2c_txn txn_list[] = {
+		{
+			.info = __func__,
+			      .addr = dev->ms5xxx_i2c_addr,
+			      .rw = PIOS_I2C_TXN_WRITE,
+			      .len = 1,
+			      .buf = &address,
+		},
+		{
+			.info = __func__,
+			      .addr = dev->ms5xxx_i2c_addr,
+			      .rw = PIOS_I2C_TXN_READ,
+			      .len = len,
+			      .buf = buffer,
+		}
+	};
+
+		rc = PIOS_I2C_Transfer(dev->dev_id, txn_list, NELEMENTS(txn_list));
+		break;
+	}
+	}
+
+	return rc;
+}
+
+/**
+* Writes one or more bytes to the MS5XXX
+* \param[in] address Register address
+* \param[in] buffer source buffer
+* \return 0 if operation was successful
+* \return -1 if dev is invalid
+* \return -2 if failed to claim SPI bus
+* \return -3 if error during transfer
+*/
+static int32_t PIOS_MS5XXX_WriteCommand(uint8_t command)
+{
+	if (PIOS_MS5XXX_Validate(dev) != 0) {
+		return -1;
+	}
+
+	int32_t rc = 0;
+
+	switch (dev->pios_bus_type) {
+	case PIOS_SPI_BUS:
```

Comment:
It will increase code complexity to conditionally include I2C and SPI, whereas the existing approach seems to work fine without the conditional includes. Is there an advantage to what you're suggesting beyond minimizing impact on F1 processors?